### PR TITLE
fix(resolvers): reliably resolve variables referenced via JS/TS file resolvers

### PR DESCRIPTION
**Pull request by czubocha**
## Summary

When a configuration value is resolved through a JavaScript or TypeScript file (for example `${file(resolver.js):export}`) and that file resolves additional variables while it runs (via `resolveVariable` / `resolveConfigurationProperty`), nested variables in the returned value could, in some edge cases, be left unresolved — returned as the literal `${...}` text instead of the resolved value.

It only happened in specific cases, and only sometimes: it depended on the order in which variables happened to resolve, so the large majority of configurations were never affected.

## What changed

A file resolver can ask the framework to resolve additional variables while the configuration itself is still being resolved. Those nested requests now join the resolution already in progress, instead of running as a separate one that could miss part of the dependency chain. As a result, dependent variables resolve consistently regardless of order.

## Impact

- Configurations that already resolved correctly are unaffected — same output, same behavior.
- Configurations that previously hit the edge case now resolve fully and consistently.
- No changes to configuration syntax or public APIs.

## Testing

- Added a regression test covering the nested-resolution case.
- Existing resolver unit and integration test suites pass.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed a regression where resolvers triggered during configuration placeholder resolution could interfere with parent resolution contexts, preventing proper handling of nested resolver invocations in complex configuration scenarios.

### CODE REVIEW

**Comment by Mmarzex** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.us.snyk.io/org/framework/pr-checks/035fb325-47e1-4be9-bceb-637ef0afb34d?refs=ca72018932847e6a874dbde23662ce95bed65991&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.us.snyk.io/org/framework/pr-checks/035fb325-47e1-4be9-bceb-637ef0afb34d/license?refs=ca72018932847e6a874dbde23662ce95bed65991&source=prChecksComment) |
| :white_check_mark: | Code Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.us.snyk.io/org/framework/project/c9cc4ca0-1caa-42d1-9aa9-df2e9b76ea90/pr-check/7aa28c63-69a8-44c6-95ed-1b4972fe9fbe?refs=ca72018932847e6a874dbde23662ce95bed65991&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: packages/sf-core/src/lib/resolvers/manager.js [hunk 1]
```diff
@@ -576,36 +576,56 @@ export class ResolverManager {
...
+              selectedPaths,
+            )
+          }
+        },
+        graph,
+        state,
+      )
+    } finally {
+      this.graphBeingProcessed = previousGraph
+      this.processingState = previousState
```
[the remark this answers is not in the document]
Comment (czubocha) [reply packages/sf-core/src/lib/resolvers/manager.js:627]: This is pre-existing limitation, not introduced by this PR
